R/phylopic_utils.R

```diff
@@ -0,0 +1,231 @@
+#' Flip a PhyloPic silhouette along its horizontal and/or vertical axis
+#'
+#' The picture can be a [Picture][grImport2::Picture-class] or png array object,
+#' e.g., from using [get_phylopic()].
+#'
+#' @param img A [Picture][grImport2::Picture-class] or png array object, e.g.,
+#'   from using [get_phylopic()].
+#' @param horizontal \code{logical}. Should the silhouette be flipped
+#'   horizontally?
+#' @param vertical \code{logical}. Should the silhouette be flipped vertically?
+#' @return A [Picture][grImport2::Picture-class] or png array object (matching
+#'   the type of `img`)
+#' @family transformations
+#' @export
+flip_phylopic <- function(img, horizontal = TRUE, vertical = FALSE) {
+  if (!is.logical(horizontal)) stop("`horizontal` must be TRUE or FALSE.")
+  if (!is.logical(vertical)) stop("`vertical` must be TRUE or FALSE.")
+  UseMethod("flip_phylopic")
+}
+
+#' @export
+flip_phylopic.Picture <- function(img, horizontal = TRUE, vertical = FALSE) {
```

Comment:
Do we want to include any error handling for `flip_phylopic` and `rotate_phylopic` if they are exported (maybe `recolour_phylopic` too)? I guess testing is already covered by calls in other function tests.